Makefile

```diff
@@ -115,24 +118,29 @@ run/registry: $(BUILD_DIR)/registry
 	TOITC_PATH=$(TOITC_PATH) TOITLSP_PATH=$(TOITLSP_PATH) SDK_PATH=$(SDK_PATH) ./$(BUILD_DIR)/registry
 
 .PHONY: image-dependencies
-image-dependencies: $(BUILD_DIR)/registry_container $(BUILD_DIR)/web_toitdocs/$(WEB_TOITDOCS_VERSION) $(BUILD_DIR)/sdk/$(SDK_VERSION) $(BUILD_DIR)/web_tpkg/$(WEB_TPKG_VERSION)
+image-dependencies: $(BUILD_DIR)/registry_container $(BUILD_DIR)/web_toitdocs $(BUILD_DIR)/sdk $(BUILD_DIR)/web_tpkg
+
+.PHONY: check-versions
+# Check that the 'VERSION' files exist and are equal to the expected versions.
+check-versions:
+	@if [ ! -f $(BUILD_DIR)/web_toitdocs/VERSION ]; then echo "Missing $(BUILD_DIR)/web_toitdocs/VERSION"; exit 1; fi
+	@if [ ! -f $(BUILD_DIR)/sdk/VERSION ]; then echo "Missing $(BUILD_DIR)/sdk/VERSION"; exit 1; fi
+	@if [ ! -f $(BUILD_DIR)/web_tpkg/VERSION ]; then echo "Missing $(BUILD_DIR)/web_tpkg/VERSION"; exit 1; fi
+	@if [ "$(WEB_TOITDOCS_VERSION)" != "$$(cat $(BUILD_DIR)/web_toitdocs/VERSION)" ]; then echo "Version mismatch in $(BUILD_DIR)/web_toitdocs/VERSION"; exit 1; fi
+	@if [ "$(SDK_VERSION)" != "$$(cat $(BUILD_DIR)/sdk/VERSION)" ]; then echo "Version mismatch in $(BUILD_DIR)/sdk/VERSION"; exit 1; fi
+	@if [ "$(WEB_TPKG_VERSION)" != "$$(cat $(BUILD_DIR)/web_tpkg/VERSION)" ]; then echo "Version mismatch in $(BUILD_DIR)/web_tpkg/VERSION"; exit 1; fi
 
 .PHONY: image
 image: image-dependencies
-	docker build --build-arg WEB_TOITDOCS_VERSION=$(WEB_TOITDOCS_VERSION) --build-arg SDK_VERSION=${SDK_VERSION} --build-arg WEB_TPKG_VERSION=${WEB_TPKG_VERSION} -t toit_registry .
+	$(MAKE) check-versions
```

Comment:
Why not just depend on check-versions?

Reply:
Primarily, because the `image-dependencies` must be run first.

Newer Makefiles have a `.WAIT` that I could add into the dependency list, but that's relatively recent and not yet supported by all GitHub builders (afaik):

```
image: image-dependencies .WAIT check-versions
```

Alternatively, I could make `check-versions` depend on `image-dependencies`.

In the end it seemed easier to just call `make` again.